src/include/duckdb/common/enums/operator_result_type.hpp

```diff
@@ -42,6 +42,22 @@ enum class OperatorFinalResultType : uint8_t { FINISHED, BLOCKED };
 //! BLOCKED means the source is currently blocked, e.g. by some async I/O
 enum class SourceResultType : uint8_t { HAVE_MORE_OUTPUT, FINISHED, BLOCKED };
 
+//! AsyncResultType is used to indicate the result of a AsyncResult, in the context of a wider operation being executed
+enum class AsyncResultType : uint8_t {
+	INVALID,  // current result is in an invalid state (eg: it's in the process of being initialized)
+	IMPLICIT, // current result depends on external context (eg: in the context of TableFunctions, either FINISHED or
+	          // ONGOING depending on output_chunk.size())
+	ONGOING,  // current result is not completed, finished (eg: in the context of TableFunctions, function accept more
+	          // iterations and might producer further results)
+	FINISHED, // current result is completed, no subsequent calls on the same state should be attempted
+	BLOCKED   // current result is blocked, no subsequent calls on the same state should be attempted, BLOCKED migth be
+	          // associated with a vector of AsycnTasks to be scheduled
```

Comment:
`might producer` -> `might produce`
`migth be` -> `might be`
`AsycnTasks` -> `AsyncTasks`

Also "might be" is too loose if I look at the code, we assert that the vector isn't empty there